### PR TITLE
FDA solution to DVARS task

### PR DESCRIPTION
**Pull request by FrancescoNottingham**
Dear @oesteban and @AnriiGegliuk,

Here is my solution to the DVARS task, any suggestions or improvements are more than welcome! 

Best,
Francesco

### CODE REVIEW

File: findoutlie/metrics.py [hunk 1]
```diff
@@ -21,12 +21,13 @@ def dvars(img):
...
-    # You may be be able to solve this in four lines, without a loop.
-    # But solve it any way you can.
-    # This is a placeholder, replace it to write your solution.
+
+    data = img.get_fdata()
+    # create two timeseries with n-1 volumes and find the difference between them
+    # remove the last volume
+    img_start = np.delete(data, slice(data.shape[-1]-1,data.shape[-1]), axis=3) 
+    # remove the first volume
+    img_end = np.delete(data, slice(0,1), axis=3)
```
Comment (oesteban) [inline findoutlie/metrics.py]: Numpy provides fancy indexing to allow doing this without actually deleting or copying the data over. Can you try to use indexing instead? TIP: I'll write one of them:
```suggestion
    img_start = np.delete(data, slice(data.shape[-1]-1,data.shape[-1]), axis=3) 
    # remove the first volume
    img_end = data[..., 1:]
```

**Review by oesteban — COMMENTED** [review]
Goes in the right direction. I made a suggestion for safer code.

File: findoutlie/metrics.py (hunk 1 above)
Comment (FrancescoNottingham) [reply findoutlie/metrics.py]: Thank you, would I change: 
```suggestion
    img_start = data[...,:-1] 
```

File: findoutlie/metrics.py [hunk 2]
```diff
@@ -21,12 +21,13 @@ def dvars(img):
...
-    # Out[2]: array([3., 6.])
-    #
-    # You may be be able to solve this in four lines, without a loop.
-    # But solve it any way you can.
-    # This is a placeholder, replace it to write your solution.
+
+    data = img.get_fdata()
+    # create two timeseries with n-1 volumes and find the difference between them
+    # remove the last volume
+    img_start = np.delete(data, slice(data.shape[-1]-1,data.shape[-1]), axis=3)
```
Comment (FrancescoNottingham) [inline findoutlie/metrics.py]: From previous comment (I accidentally closed), suggested change
```suggestion
    img_start = data[...,:-1]
```

**Pull request merged**